cornflow/endpoints/case.py

```diff
@@ -315,3 +318,56 @@ def post(self, idx):
         validate_and_continue(marshmallow_obj(), payload["data"])
 
         return self.post_list(payload)
+
+
+class CaseCompare(MetaResource, MethodResource):
+    """
+    Endpoint used to generate the json patch of two given cases
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.model = CaseModel
+        self.query = CaseModel.get_all_objects
+        self.primary_key = "id"
+
+    @doc(
+        description="Compares the data and / or solution of two given cases",
+        tags=["Cases"],
+    )
+    @Auth.auth_required
+    @marshal_with(CaseCompareResponse)
+    @use_kwargs(QueryCaseCompare, location="query")
+    def get(self, idx1, idx2, **kwargs):
+        """
+        API method to generate the json patch of two cases given by the user
+        It requires authentication to be passed in the form of a token that has to be linked to
+        an existing session (login) made by a user
+
+        :param int idx1: ID of the base case for the comparison
+        :param int idx2: ID of the case that has to be compared
+        :return:an object with the instance or instance and execution ID that have been created and the status code
+        :rtype: Tuple (dict, integer)
+        """
+        if idx1 == idx2:
+            raise InvalidData("The case identifiers should be different", 400)
+        case_1 = self.model.get_one_object_from_user(self.get_user(), idx1)
+        case_2 = self.model.get_one_object_from_user(self.get_user(), idx2)
+
+        if case_1 is None:
+            raise ObjectDoesNotExist("The first case does not exist")
```

Comment:
I prefer something like "You do not have access to the first case". I'm not sure if we should give back the code of the case...

Reply:
It does not have to mean that you do not have access, it can also mean that the case does not exist and the id is wrong. But I see your point that it could be both so I should change the message